src/components/ListItem.jsx

```diff
@@ -14,24 +20,54 @@ export function ListItem({ listPath, item, name }) {
 	useEffect(() => {
 		if (item.dateLastPurchased) {
 			const isExpired = itemIsExpired(item);
-
 			if (isExpired) {
 				updateItem(listPath, item, true);
 			}
 
 			setIsChecked(!isExpired);
 		}
-	}, [item.dateLastPurchased]);
+	}, [item]);
+
+	const getPurchaseUrgency = (item) => {
+		const today = Timestamp.now();
+		const numberOfDays = getDaysBetweenDates(today, item.dateNextPurchased);
+
+		const isInactive = itemIsInactive(item);
+		const isOverdue = itemIsOverdue(item);
+		const daysPastDue =
+			isOverdue && getDaysBetweenDates(item.dateNextPurchased, today);
+
+		switch (true) {
+			case isInactive:
+				return 'inactive (last purchased over 60 days ago)';
+			case isOverdue:
+				return `overdue (${daysPastDue === 1 ? '1 day past' : `${daysPastDue} days past`})`;
+			case numberOfDays <= 7:
+				return `soon (in ${numberOfDays === 1 ? '1 day' : `${numberOfDays} days`})`;
+			case numberOfDays > 7 && numberOfDays < 30:
+				return `kind of soon (in ${numberOfDays} days)`;
+			case numberOfDays >= 30:
+				return `not soon (in ${numberOfDays} days)`;
```

Comment:
If you plan to keep the text in the parenthesis, you may consider writing the urgency details below the item. I think this app would mostly be used on a mobile device, and right now, the mobile view is a bit tricky with all this information in one line. 
<img width="388" alt="Screenshot 2024-03-15 at 6 30 41 PM" src="https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/97368143/65627bbd-e33c-463b-8bec-78335152da00">

Reply:
Also, in this screenshot you'll notice there's an empty item. But I'm assuming it was created in firebase, as we have implemented logic against this behavior.